app/questionnaire_state/state_question.py

```diff
@@ -6,3 +8,54 @@ def __init__(self, id, schema_item):
         super().__init__(id=id, schema_item=schema_item)
         self.answers = []
         self.children = self.answers
+
+    def remove_answer(self, answer):
+        if answer in self.answers:
+            self.answers.remove(answer)
+
+    def update_state(self, user_input):
+
+        if self.schema_item.type == 'RepeatingAnswer':
+            state_answers = self.children[:]
+
+            for child in state_answers:
+                answer_keys = [key if key.startswith(child.schema_item.id) else None for key in user_input]
+                answer_instances = sorted(list(filter(None, answer_keys)))
+                num_instances = len(answer_instances)
+
+                if num_instances > 1:
```

Comment:
I'll give it a test it might be okay to remove it after all. Confusing myself because it's changed to build the state on update_state now rather than construct state.